### ISSUE TITLE
Calendar colors aren't being displayed

### ISSUE THREAD

**Issue by samdmarshall** [pimutils/khal#542]
I have khal setup and configured with my own iCloud-based calendar and two read-only calendar subscriptions, each with their own calendar colors. However, passing the `--colors` flag on startup of both `ikhal` and `khal` I don't see those colors get used for events for the respective calendars. My setup is running from within fish shell (2.4.0) in Terminal.app on OS X 10.12. If there is any additional information I can provide about this to help debug what is going on; i would be happy to.

Thanks for making such a great piece of software!

**Comment by hobarrera** [comment]
Have you run `vdirsyncer metasync`? (this sync the calendar metadata, which is not normally done).

If so, does you vdir contain a file named `color`?

**Comment by geier** [comment]
You are welcome (it would obviously be even greater software, if it would work properly)!

I'm not sure I understood what isn't working here, are we talking about the color of the events, or the colored days (in the calendar)?

In the former case, please make sure that `highlight_event_days` is set to true (in the `[default]` section), in the later, please (before you actually check @hobarrera suggestions above) have a look at the output of `khal -v calendar` and check if the colors were correctly understood by khal in the first place.

**Comment by samdmarshall** [comment]
I have run `vdirsyncer metasync` and that didn't seem to do anything. I have given the `khal -v calendar` a try and i see that it is printing out the setting `color = None` for all of my calendars. I have configured each calendar with a color in the `khal.conf`, so I presume this is a bug.

I added the `highlight_event_days` key as suggested, and that is not causing any colors either, but I suspect this is due to khal thinking that there are no colors to show.

**Comment by geier** [comment]
Which version of khal are you running? (`khal --version` should tell),
and can you share the relevant part of your configuration file?


Quoting Samantha Marshall (2016-12-16 19:59:10)

**Comment by samdmarshall** [comment]
Version: `khal, version 0.8.4`

Config:
```
[default]
highlight_event_days = true

[calendars]
[[subscription_1]]
path = ~/Calendars/subscription_1
type = discover
color = dark red
readonly = True
[[read_write_calendar]]
path = ~/Calendars/read_write_calendar
type = discover
color = dark magenta
[[subscription_2]]
path = ~/Calendars/subscription_2
type = discover
color = dark green
readonly = True

[locale]
timeformat = %H:%M
dateformat = %m.%d.%Y
longdateformat = %m.%d.%Y
datetimeformat = %m.%d.%Y %H:%M
longdatetimeformat = %m.%d.%Y %H:%M

[view]
frame = True
```

**Comment by geier** [comment]
This probably is a bug that we recently fixed on the master branch, where the color settings were disregarded when `type=discover`.

You have several options here: 
* upgrade (by hand) to the latest version of khal
* split up the discover calendars (in the configuration file) into separate sections, when `type=caldav` (or unset) the color setting should work as intended
* create files named `color` in each vdir with the desired color, but this might cause problems when syncing

**Comment by samdmarshall** [comment]
I'd be happy to try to build this locally and see if it works, however I have version 0.8.4 via Homebrew and building from master nets me 0.8.2. Which branch/configuration should i be trying this with?

**Comment by geier** [comment]
Building from master should give you a version like this:
0.8.2.dev304+ngd6f08a4 (if you cloned the repository from github, if you
downloaded a tgz, it might be different) this is in fact the latest
version, but the version name gets generated from the last tag on that
same branch (and 0.8.3 and 0.8.4 where done from a different branch).

Quoting Samantha Marshall (2016-12-18 21:25:25)

**Comment by samdmarshall** [comment]
OK, I have (some) colors working, I had to add the additional line to my config file for the pair which was very unclear from the documentation:

``
metadata = ["color"]
```

However I still cannot get readonly calendars to have an associated color to be displayed.

**Comment by geier** [comment]
The metadata settings is for vdirsyncer's config, right?  And it really shouldn't be needed, because you set the colors explicitly in your configuration file.

I just noticed something else in your configuration file: 

In case you are now using the latest version of khal (master branch): the way `discover` works changed, it now doesn't recursively decent into that directory, but uses globbing (and yes, the docs really need some work).

So if your read-only calendars are `~/Calendars/subscription_1/cal1/` and `~/Calendars/subscription_1/cal2/`, the relevant section should look like this:

```ini
[[subscription_1]]
path = ~/Calendars/subscription_1/*
type = discover
color = dark red
readonly = True
```

Could you perhaps run `khal -v` again and paste the output (remove all sensitive lines)? If I can't figure it out then, I guess I'll need to put in some more debugging output.

While less concise, you could probably work with this 

```ini
[[cal1]]
path = ~/Calendars/subscription_1/cal1/
color = dark red
readonly = True

[[cal2]]
path = ~/Calendars/subscription_1/cal2/
color = dark red
readonly = True
```

instead. If this doesn't work, I really have no idea what's going on.

**Comment by samdmarshall** [comment]
With some experimenting i was able to get colors to be picked up by khal, but only by creating a `color` file in the directory with a hex color code in it (as if it were created by `vdirsyncer`). This seems to do what I want, even though the colors specified in the `khal.conf` file are being completely ignored still. 

I am in an `xterm-256color` terminal, yet `ikhal` or `khal interactive` nets me color while `khal` (even with `--color`) only shows me various shades of gray.

**Comment by untitaker** [comment]
If vdirsyncer didn't create those files when intended, please file a bug against vdirsyncer. I recently added a [short tutorial for color sync ](https://vdirsyncer.pimutils.org/en/stable/tutorial.html#metadata-synchronization), if there's anything unclear about that one (or if it's hard to find) please file bugs as well.

**Comment by samdmarshall** [comment]
I don't think it is a fault of `vdirsyncer` that those files weren't created, as read-only "http"-type calendar subscriptions apparently don't support metadata syncing. I am apparently fooling `khal` into loading colors that it isn't loading from the config for whatever reason.

@geier When I ran `khal -v` all the color values were displayed as set to `None` even though it was loading the config file I posted earlier in the thread. Nothing about that config has changed.
